benchmarks/torchvision_classification/train.py

```diff
@@ -0,0 +1,346 @@
+
+    elif args.ds_type == "mapstyle":
+        builder = helpers.PreLoadedMapStyle if args.preload_ds else torchvision.datasets.ImageFolder
+        train_dataset = builder(train_dir, transform=train_preset)
+        val_dataset = builder(val_dir, transform=val_preset)
+
+        train_sampler = torch.utils.data.distributed.DistributedSampler(train_dataset, shuffle=True)
+        val_sampler = torch.utils.data.distributed.DistributedSampler(val_dataset, shuffle=False)
+        train_shuffle = None  # but actually True
+
+    else:
+        raise ValueError(f"Invalid value for args.ds_type ({args.ds_type})")
+
+    data_loader_arg = args.data_loader.lower()
+    if data_loader_arg == "v1":
+        train_data_loader = torch.utils.data.DataLoader(
+            train_dataset,
+            batch_size=args.batch_size,
+            shuffle=train_shuffle,
+            sampler=train_sampler,
+            num_workers=args.workers,
+            pin_memory=True,
+            drop_last=True,
+        )
+        val_data_loader = torch.utils.data.DataLoader(
+            val_dataset,
+            batch_size=args.batch_size,
+            sampler=val_sampler,
+            num_workers=args.workers,
+            pin_memory=True,
+        )
+    elif data_loader_arg == "v2":
+        if args.ds_type != "dp":
+            raise ValueError("DataLoader2 only works with datapipes.")
+
+        # Note: we are batching and collating here *after the transforms*, which is consistent with DLV1.
+        # But maybe it would be more efficient to do that before, so that the transforms can work on batches??
+
+        train_dataset = train_dataset.batch(args.batch_size, drop_last=True).collate()
+        train_data_loader = DataLoader2(
+            train_dataset,
+            datapipe_adapter_fn=adapter.Shuffle(),
+            reading_service=MultiProcessingReadingService(num_workers=args.workers),
+        )
+
+        val_dataset = val_dataset.batch(args.batch_size, drop_last=True).collate()  # TODO: Do we need drop_last here?
+        val_data_loader = DataLoader2(
+            val_dataset,
+            reading_service=MultiProcessingReadingService(num_workers=args.workers),
+        )
+    else:
+        raise ValueError(f"invalid data-loader param. Got {args.data_loader}")
+
+    return train_data_loader, val_data_loader, train_sampler
+
+
+def main(args):
+    if args.output_dir:
+        utils.mkdir(args.output_dir)
+
+    utils.init_distributed_mode(args)
+    print("\n".join("%s: %s" % (k, str(v)) for k, v in sorted(dict(vars(args)).items())))
+
+    device = torch.device(args.device)
+
+    if args.use_deterministic_algorithms:
+        torch.backends.cudnn.benchmark = False
+        torch.use_deterministic_algorithms(True)
+    else:
+        torch.backends.cudnn.benchmark = True
+
+    train_data_loader, val_data_loader, train_sampler = create_data_loaders(args)
+
+    num_classes = 1000  # I'm lazy. TODO change this
```

Comment:
What does this do? Define the number of classes to classify?

Reply:
Yes, we need to pass this to the model's constructor. It needs to know how many classes there are because this directly impact the number of outputs of the model (one output (probability) per class).